test/e2e/nodeconditions.go

```diff
@@ -0,0 +1,249 @@
+	// is the default that is set today. This test might break if the default value changes. This value
+	// can be configured by setting the "low-diskspace-threshold-mb" flag while starting a kubelet.
+	// However, kubelets are started as part of the cluster start up, once, before any e2e test is run,
+	// and remain unchanged until all the tests are run and the cluster is brought down. Changing the
+	// flag value affects all the e2e tests. So we are hard-coding this value for now.
+	lowDiskSpaceThreshold uint64 = 256 * mb
+
+	nodeconditionsOODTimeOut = 1 * time.Minute
+
+	numNodeConditionsPods = 3
+
+	unfilledNodeIdx = 0
+)
+
+type podId struct {
+	name  string
+	value string
+}
+
+// Plan:
+// 1. Fill disk space on all nodes except one. One node is left out so that we can schedule pods
+//    on that node. Arbitrarily choose that node to be node with index 0.
+// 2. Get the CPU capacity on unfilled node.
+// 3. Divide it into number of pods we want to schedule - 1. We want to schedule 3 pods, so
+//    divide CPU capacity by 2.
+// 4. Request the divided capacity for each pod.
+// 5. Attempt to schedule the final pod and observe that it fails.
+//
+// Reason: although there is CPU available on other nodes, pods are not schedulable on those nodes
+//         because the other nodes are out of disk.
+//
+// TODO(madhusudancs): Today, we use CPU instead of directly using storage as resource constraint
+// because storage is still not a schedulable resource. Switch to using api.ResourceStorage when
+// storage becomes a schedulable resource.
+var _ = Describe("NodeConditions", func() {
+	var c *client.Client
+	framework := Framework{BaseName: "node-conditions"}
+
+	BeforeEach(func() {
+		framework.beforeEach()
+		c = framework.Client
+
+		// Fill disk space on all the nodes except the first node (index 0).
+		fillDiskSpace(c)
+	})
+
+	AfterEach(func() {
+		defer framework.afterEach()
+		recoverDiskSpace(c)
+	})
+
+	It("runs out of disk space", func() {
+		// Get all nodes, and kick off the test on each.
+		nodelist, err := listNodes(c, labels.Everything(), fields.Everything())
+		if err != nil {
+			Failf("Error getting nodes: %v", err)
+		}
+		unfilledNode := nodelist.Items[unfilledNodeIdx]
+		By(fmt.Sprintf("Get CPU capacity on node %s", unfilledNode.Name))
+
+		milliCPU := unfilledNode.Status.Capacity.Cpu().MilliValue()
+		// Per pod CPU should be just enough to fit only (numNodeConditionsPods - 1) pods on the
+		// given node. We compute this value by dividing the available CPU capacity on the given
+		// node by (numNodeConditionsPods - 1) and subtracting ϵ from it.
+		podCPU := (milliCPU / (numNodeConditionsPods - 1)) - (milliCPU / 10)
+
+		ns := framework.Namespace.Name
+		podClient := c.Pods(ns)
+
+		By("Creating pods")
+
+		podIds := make([]podId, 0)
+
+		for i := 0; i < numNodeConditionsPods; i++ {
+			id := podId{
+				name:  fmt.Sprintf("pod-nodecondition-%d", i),
+				value: strconv.Itoa(time.Now().Nanosecond()),
```

Comment:
I was using it in the loop below to check whether a pod was assigned to a node. But I can do that without labels, by just calling podClient.Get() instead of podClient.List().

Removed labels entirely. And the podId struct that came with it. Simplifies the code a little bit.